lib/jxl/enc_params.h

```diff
@@ -254,6 +254,19 @@ struct CompressParams {
   size_t ec_resampling = 1;
   // Skip the downsampling before encoding if this is true.
   bool already_downsampled = false;
+
+  void PostInit() {
+    if (resampling == 0) {
```

Comment:
why not "== 0 || == -1" and the C API just sets the value?

Can we add a test?

Reply:
We can make it signed too :)

As for tests: mostly that setting it to -1 works and doesn't crash/...

Reply:
Made it signed, and also made it -1 by default, because we claim in the documentation that -1 is the default but actually 1 was the default.

I don't think a separate test is needed now, because all the existing tests that don't explicitly change it will now have it set to -1.